packages/rich-text/src/component/use-input-and-selection.js

```diff
@@ -275,29 +275,10 @@ export function useInputAndSelection( props ) {
 				return;
 			}
 
-			if ( ! isSelected ) {
-				// We know for certain that on focus, the old selection is invalid.
-				// It will be recalculated on the next mouseup, keyup, or touchend
-				// event.
-				const index = undefined;
-
-				record.current = {
-					...record.current,
-					start: index,
-					end: index,
-					activeFormats: EMPTY_ACTIVE_FORMATS,
-				};
-				onSelectionChange( index, index );
-			} else {
+			if ( isSelected ) {
 				applyRecord( record.current );
 				onSelectionChange( record.current.start, record.current.end );
 			}
-
-			// Update selection as soon as possible, which is at the next animation
-			// frame. The event listener for selection changes may be added too late
-			// at this point, but this focus event is still too early to calculate
-			// the selection.
-			rafId = defaultView.requestAnimationFrame( handleSelectionChange );
```

Comment:
Should we first try to remove it without removing this?

Reply:
It's not clear to me why we need this since when "focus" happens, the selection changes so the callback is called no?

Reply:
I'm just curious which of these changes makes which tests fail, so we can address it more easily.

Reply:
I'm just curious which of these changes makes which tests fail, so we can address it more easily.

Reply:
I'm just curious which of these changes makes which tests fail, so we can address it more easily.

Reply:
Looks like everything passes now that I restored that part

Reply:
I'd say let's look into that separately then.